### PR TITLE
Spec gloss vs metal rough

### PR DESCRIPTION
**Pull request by emackey**
This adds the "two water bottles" side-by-side comparison of spec/gloss and metal/rough as a new sample model in its own right.  New labels were added, and they too are part of the test.

/cc #181 and AnalyticalGraphicsInc/cesium#7006

/cc @bghgary @OmarShehata @abwood

### CODE REVIEW

**Comment by emackey** [comment]
See the [README](https://github.com/KhronosGroup/glTF-Sample-Models/blob/f72fc12a44323f0d65f5220de2426bff4abef116/2.0/SpecGlossVsMetalRough/README.md)

**Comment by lexaknyazev** [comment]
"License Information" part of the readme contains only authors, not usage / distribution license.

**Comment by emackey** [comment]
Yes, this is because it is derived from the Microsoft WaterBottle, which similarly lists authors but not a license.  @bghgary or @sbtron can a real license be added to WaterBottle and friends?  Preferably one that permits derivative works!

**Comment by bghgary** [comment]
I'll let Patrick Ryan answer this in more detail. I think we wanted to avoid putting any license because it still caused restrictions even with the most permissive one. We wanted this to be completely open for use. @PatrickRyanMS what do you think?

**Comment by lexaknyazev** [comment]
Lack of license usually means that nothing could be done with the model without explicit permission.
To give it away completely, the typical way is to declare it "public domain" (if applicable) or use something like CC0.

**Comment by PatrickRyanMS** [comment]
I agree with @bghgary that we want these assets publicly distributable and open for anyone to use, remix, share, etc. The issue with even applying a CC-BY 4.0 license is that since we work for Microsoft and these contributions are part of our work for the company, I don't know that we can just assign a license, even CC-BY, without getting our legal department to sign off on us doing that. This is a question that we will need to dig a little further on before we can attach any legal language to the assets, but we do want them to be public domain.

**Comment by emackey** [comment]
@bghgary @PatrickRyanMS Thanks, and please do follow up on this.  @lexaknyazev is correct that legally we shouldn't be doing anything with the Microsoft-supplied models, not even including them for distribution with the sample models, without a license.

**Comment by cx20** [comment]
@emackey I added this model to [gltf-test](https://github.com/cx20/gltf-test).

I think the current status is as follows.
https://github.com/cx20/gltf-test#extension-test-models
image

| ||
|:------:|:------:|
|[image](https://cx20.github.io/gltf-test/examples/threejs/index.html?category=tutorialModels&model=SpecGlossVsMetalRough&scale=10&type=glTF)|[image](https://cx20.github.io/gltf-test/examples/babylonjs/index.html?category=tutorialModels&model=SpecGlossVsMetalRough&scale=10&type=glTF)|
|[image](https://cx20.github.io/gltf-test/examples/xeogl/index.html?category=tutorialModels&model=SpecGlossVsMetalRough&scale=10&type=glTF)|[image](https://cx20.github.io/gltf-test/examples/Hilo3d/index.html?category=tutorialModels&model=SpecGlossVsMetalRough&scale=10&type=glTF)|
|[image](https://cx20.github.io/gltf-test/examples/playcanvas/index.html?category=tutorialModels&model=SpecGlossVsMetalRough&scale=10&type=glTF)|[image](https://cx20.github.io/gltf-test/examples/czpg/index.html?category=tutorialModels&model=SpecGlossVsMetalRough&scale=10&type=glTF)|

**Comment by emackey** [comment]
@cx20 Thanks.  I didn't realize the spec/gloss ecosystem was in such bad shape.  Cesium support is coming soon.  Some of the others appear to be really close, like ThreeJS and Hilo3D.

**Comment by emackey** [comment]
@bghgary @PatrickRyanMS How should we proceed with this one?  Did you reach out to legal, or is there any other option?

**Comment by bghgary** [comment]
@sbtron is going over this with legal which takes some time. @sbtron, any status on this?

**Comment by emackey** [comment]
OK, in the master branch, Microsoft has added CC0 licenses to its sample models (thanks so much @bghgary and @sbtron!).  I've copied that license here to this new model, since it's just a re-packaging of two different versions of the existing water bottle model side-by-side.

This should be good to merge now.

File: 2.0/SpecGlossVsMetalRough/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+No fallback to metal/rough is provided for the spec/gloss side, so unlike most other samples, `KHR_materials_pbrSpecularGlossiness` extension support is explicitly required by this model.  The model is expected to fail to load on viewers that do not support the extension, as such support is not required by the core glTF specification.  This model marks the extension under `extensionsRequired`, meaning any viewer that doesn't support the extension should reject the model completely.  So, a complete failure to load the model, possibly with a message about lack of support for the extension, is not considered a test failure for core glTF 2.0.
+
+However, if the model loads into a viewer, and the two water bottles don't look almost exactly the same, then that does constitute a test failure.
+
+The labels floating above the water bottles should also have the same visual appearance (other than the text of the label being different).  One should not be shiny or more reflective than the other.
+
+## License Information
+
+Based on the Water Bottle sample model, donated by Microsoft for glTF testing.
```
Comment (bghgary) [inline 2.0/SpecGlossVsMetalRough/README.md]: I don't think it's appropriate to say "donated by Microsoft for glTF testing" here?

File: 2.0/SpecGlossVsMetalRough/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+[![CC0](http://i.creativecommons.org/p/zero/1.0/88x31.png)](http://creativecommons.org/publicdomain/zero/1.0/)
```
Comment (bghgary) [inline 2.0/SpecGlossVsMetalRough/README.md]: Add two spaces at the end of the line to put the text on a separate line.

File: 2.0/SpecGlossVsMetalRough/README.md (hunk 1 above)
Comment (emackey) [reply 2.0/SpecGlossVsMetalRough/README.md]: Removed.

File: 2.0/SpecGlossVsMetalRough/README.md (hunk 2 above)
Comment (emackey) [reply 2.0/SpecGlossVsMetalRough/README.md]: Fixed.  I had copy/pasted this, but I have my copy of VSCode configured to remove trailing spaces automatically.  Sad that markdown places significance on such things.

**Comment by cx20** [comment]
@emackey @donmccurdy I have doubts about the display result of Three.js.
The Metallic Roughness model reflects the environment map, but the Specular Glossiness model does not seem to reflect the environment map. Is this a correct display result?

[Three.js + SpecGlossVsMetalRough.gltf result:](https://cx20.github.io/gltf-test/examples/threejs/index.html?category=tutorialModels&model=SpecGlossVsMetalRough&scale=10&type=glTF)
image

**Comment by donmccurdy** [comment]
Thanks again for these side-by-side comparisons. ❤️

It's entirely possible the spec/gloss result is incorrect in three.js, I'll file a bug.

**Pull request merged**